### ISSUE TITLE
Advanced color selector

### ISSUE THREAD

**Issue by caryoscelus** [mbasaglia/Qt-Color-Widgets#29]
So as mentioned earlier, i'm developing an advanced color selector widget (which could be used as dock in graphical applications, so more compact than color dialog), which would be basically a configurable composite widget containing various existing widgets (wheel, 2d slider, perhaps multiple 1d sliders, etc).

Planned features so far:

- [x] switching between few primary color selector forms (triangle & square wheel and 2d+hue sliders for now)
- [ ] harmony color selection (kinda done, but no UI to actually make them active for now)
- [ ] configurable sliders
- [ ] color history
- [ ] foreground/background color switching

Since i've started implementing this, i've run into some issues for which i'm going to open separate threads (so that we can keep this one just for feature and maybe implementation discussion).

Here's my development branch: https://github.com/caryoscelus/Qt-Color-Widgets/tree/advanced_color_selector

**Comment by mbasaglia** [comment]
Just an idea, but "color history" could use the palette widget.
"foreground/background color switching" could be done as an extension of the split color preview (it would just need a slot to switch the main and comparison color).

Also, dunno if it's useful but I've written this compound color editor some time ago as part of another project: https://github.com/mbasaglia/PixelCayman/blob/master/src/ui/widgets/color_editor.hpp

**Comment by caryoscelus** [comment]
>"color history" could use the palette widget.

Yeah, i thought about that too (though i haven't looked at palette widget yet).

>"foreground/background color switching" could be done as an extension of the split color preview

Would probably work for beginning, but later it might be better to create a more visually rich one (such as the typical overlapping squares/circles).

**Comment by nicholascarroll** [comment]
The Natron project (https://github.com/MrKepzie/Natron) needs one of these.

**Comment by mbasaglia** [comment]
@nicholascarroll which one?

**Comment by nicholascarroll** [comment]
I think it should be a simple colour wheel because when adjusting the colour the user look at the image not so much at the colour wheel. Here is the issue for it https://github.com/MrKepzie/Natron/issues/1592.
I sort of put my hand up to code it but my C++ skills are practically nonexistent.
